miio/device.py

```diff
@@ -143,6 +145,8 @@ def send(
         :param int retry_count: How many times to retry on error
         :param dict extra_parameters: Extra top-level parameters
         """
+        if hasattr(self, "retry_count"):
+            retry_count = self.retry_count
```

Comment:
This would override the user-given parameter which is a no-no.

A workable solution for now would be to move these settings to be class variables for `Device`:
```
class Device:
    timeout = 5
    retry_count = 3
```
For individual methods (`__init__()` and this), the corresponding parameters should default to `None` instead of their current values. If those parameters are not filled by the caller, the fallback is to use the class-defined values.

That way this piece of code would became and the current behavior for downstream users stays intact:
```
retry_count = retry_count if retry_count is not None else self.retry_count
```

Could you also please add some tests for this?

Reply:
> Hi @titilambert, first of all, your efforts to provide tests is really appreciated! :-) I saw some commits on this PR in notifications, and wanted to mention that you can also run tests locally using `pytest miio` in the main source directory, or by calling `tox` to run all checks what the CI system does.
> 
> I noticed you are trying to check out if a parameter has been given (rightly so!), so I just wanted to mention that you may want to check if `assert_called_with()` works instead of manually call args list. I'm not that familiar with pytest/mock myself either, so I cannot definitely say if that works or not, but the name implies that it'd be useful in this case.

@rytilahti  Thanks! I just found that function :D I just pushed a working commit, and I just squashed all my tests commits ...

It's now ready to review